executor/brie.go

```diff
@@ -106,53 +221,96 @@ var globalBRIEQueue = &brieQueue{
 func (bq *brieQueue) registerTask(
 	ctx context.Context,
 	info *brieTaskInfo,
-) (context.Context, uint64) {
+	s sqlexec.RestrictedSQLExecutor,
+) (context.Context, uint64, *brieQueueItem, error) {
 	taskCtx, taskCancel := context.WithCancel(ctx)
 	item := &brieQueueItem{
-		info:   info,
-		cancel: taskCancel,
-		progress: &brieTaskProgress{
-			cmd:   "Wait",
-			total: 1,
-		},
+		info:     info,
+		cancel:   taskCancel,
+		progress: &brieTaskProgress{},
 	}
-
-	taskID := atomic.AddUint64(&bq.nextID, 1)
-	bq.tasks.Store(taskID, item)
-
-	return taskCtx, taskID
+	item.progress.lock.total = 1
+	item.progress.setCmd(cmdWait)
+
+	// use base64 encode to avoid SQL injection
+	sql := fmt.Sprintf(`INSERT INTO mysql.brie_tasks (
+				kind, origin_sql, queue_time, data_path, conn_id, status, progress, cancel
+			) VALUES ('%s', '%s', '%s', '%s', %d, '%s', %f, %d);`,
+		info.kind.String(),
+		base64.StdEncoding.EncodeToString([]byte(info.originSQL)),
+		info.queueTime.String(),
+		base64.StdEncoding.EncodeToString([]byte(info.storage)),
+		info.connID,
+		item.progress.getCmd(),
+		item.progress.getFraction(),
+		0,
+	)
```

Comment:
using `format.OutputFormat` from `util/format.go` should be sufficient to protect against SQL injection, no need to use base64.